packages/client/src/views/CorrectionForm/CorrectionSummary.tsx

```diff
@@ -172,6 +151,26 @@ class CorrectionSummaryComponent extends React.Component<IFullProps, IState> {
       declaration: { event },
       userRole
     } = this.props
+
+    const group = {
+      ...this.group,
+      fields: replaceInitialValues(
+        this.group.fields,
+        this.props.declaration.data[this.section.id] || {},
+        this.props.declaration.data
+      )
+    }
+
+    const currency = getCurrencySymbol(
+      this.props.offlineResources.config.CURRENCY
+    )
+
+    ;(group.fields[0].nestedFields as any).REQUIRED[0].label.defaultMessage =
+      this.props.intl.formatMessage(
+        messages.correctionSummaryTotalPaymentLabel,
+        { currency }
+      )
```

Comment:
This doesn't look like the best way of doing this: 
- Mutating objects (especially if the data originates in the redux store) is can cause difficult-to-solve bugs
- Default message is just that – a default message that shouldn't be dynamic
- The property access field here is long and there are many embedded assumptions – if one of them is not true or like you imagined in production, it will throw an exception

Where does this label get rendered in the code? Can currency be supplied from there? 

As a rule of thumb, copy and formatting logic should be done right at the rendering point and not earlier than that.

Maybe @tahmidrahman-dsi or @Zangetsu101 could help you on this